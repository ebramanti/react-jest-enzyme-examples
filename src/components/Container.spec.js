import React from "react";
import { shallow, mount } from "enzyme";
import { Container } from "./Container";

describe("Container", () => {
    it("renders a default posts page", () => {
        const wrapper = mount(<Container/>);
        expect(wrapper).toMatchSnapshot();
    });

    describe("#addPost", () => {
        it("adds post to top-level state", () => {
            const wrapper = shallow(<Container/>);
            wrapper.instance().addPost({ title: "New", body: "Post" });
            expect(wrapper.state("posts")).toHaveLength(2);
        });
    });

    describe("#deletePost", () => {
        it("deletes post from top-level state", () => {
            const wrapper = shallow(<Container/>);
            wrapper.instance().deletePost(0);
            expect(wrapper.state("posts")).toHaveLength(0);
        });
    });
});
