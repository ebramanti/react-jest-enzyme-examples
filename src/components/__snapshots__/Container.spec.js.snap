// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container renders a default posts page 1`] = `
<Container>
  <NewPost
    key="new-post"
    onPostCreated={[Function]}
  >
    <table>
      <tbody>
        <tr>
          <td>
            <input
              onChange={[Function]}
              type="text"
              value=""
            />
          </td>
          <td>
            <input
              onChange={[Function]}
              type="text"
              value=""
            />
          </td>
          <td>
            <button
              onClick={[Function]}
            >
              Create Post
            </button>
          </td>
        </tr>
      </tbody>
    </table>
  </NewPost>
  <PostsList
    key="posts-list"
    onPostDeleted={[Function]}
    posts={
      Array [
        Object {
          "body": "World",
          "title": "Hello",
        },
      ]
    }
  >
    <table>
      <tbody>
        <Post
          body="World"
          index={0}
          key="Hello-World-0"
          onPostDeleted={[Function]}
          title="Hello"
        >
          <tr>
            <td>
              Hello
            </td>
            <td>
              World
            </td>
            <td>
              <button
                id="reverse"
                onClick={[Function]}
              >
                Reverse Title
              </button>
            </td>
            <td>
              <button
                id="delete"
                onClick={[Function]}
              >
                Delete Post
              </button>
            </td>
          </tr>
        </Post>
      </tbody>
    </table>
  </PostsList>
</Container>
`;
